addons/mail/static/src/core/common/store_service.js

```diff
@@ -563,15 +591,15 @@ export class Store extends BaseStore {
     }
 
     async joinChat(id, forceOpen = false) {
-        const { channel_id, data } = await rpc("/discuss/channel/get_or_create_chat", {
-            partners_to: [id],
-        });
-        this.store.insert(data);
-        const thread = this.store.Thread.get({ id: channel_id, model: "discuss.channel" });
+        const { channel } = await this.fetchStoreData(
+            "/discuss/get_or_create_chat",
+            { partners_to: [id] },
+            { readonly: false, requestData: true }
+        );
         if (forceOpen) {
-            await thread.openChatWindow({ focus: true });
```

Comment:
Note: this appears to be wrong, in discuss app we would want to not open the chat window.